### ISSUE TITLE
DI fails with AOT

### ISSUE THREAD

**Issue by GochoMugo** [GochoMugo/ionic-gallery-modal#1]
#### reproduction:

In your terminal:

```bash
# Clone RR (see references at the bottom).
$ git clone https://github.com/GochoMugo/ionic-gallery-modal-BR-1.git br1

# Change directory.
$ cd br1

# Install deps.
$ npm install

# Build with AoT compiler.
$ npm run build -- --aot

# Serve the files.
$ npx http-server www -a localhost -p 8100 -c -1
```

Browser console output (on visiting the site):

```
ERROR Error: Uncaught (in promise): Error: No provider for ViewController!
````

#### references:

* **Reproduction repo (RR):** https://github.com/GochoMugo/ionic-gallery-modal-BR-1

**Comment by GochoMugo** [comment]
##### bug:

`ionic-angular` is listed under `dependencies` in the main package
manifest i.e. `/package.json` (from root of repo). During `npm install`ing
this library, a separate `ionic-angular` is re-installed at `node_modules/ionic-gallery-modal/node_modules/ionic-angular/`. AoT fails as it distinguishes the different classes from the duplicate dependencies.

##### fix:

Move peer-dependencies to the main package manifest too.

**Comment by GochoMugo** [comment]
Fixed in **v0.3.1**.
